src/components/resource/show.js

```diff
@@ -250,14 +265,23 @@ export default class ResourceShow extends Component {
                   data-test-eholdings-resource-show-selected
                   htmlFor="resource-show-toggle-switch"
                 >
-                  <h4>{resourceSelected ? 'Selected' : 'Not selected'}</h4>
+                  {
+                    model.update.isPending ? (
+                      <Icon icon='spinner-ellipsis' />
+                    ) : (
+                      <h4>{resourceSelected ? 'Selected' : 'Not selected'}</h4>
+                    )
+                  }
                   <br />
-                  <ToggleSwitch
-                    onChange={this.handleSelectionToggle}
-                    checked={model.destroy.isPending ? false : resourceSelected}
-                    isPending={model.destroy.isPending || isSelectInFlight}
-                    id="resource-show-toggle-switch"
-                  />
+                  { ((!resourceSelected && !isSelectInFlight) || (!this.props.model.isSelected && isSelectInFlight)) && (
+                    <Button
+                      buttonStyle="primary"
+                      onClick={this.handleHoldingStatus}
+                      disabled={model.destroy.isPending || isSelectInFlight}
+                      data-test-eholdings-resource-add-to-holdings-button
+                    >
+                      Add to holdings
```

Comment:
Does this work nicely or is it too much loading status happening? 
`{isSelectInFlight ? 'Adding' : 'Add to holdings'}` 
instead of just
`Add to holdings`

Reply:
I don't think it will add nor take away anything. We have the loading ellipsis and the disabling of the button happening. If the button remained blue then i would say yeah the `adding` text would help. But since is turns gray that is why i am on the fence here.

Reply:
Cool, I'll defer to your instinct on that one.